File: crates/npt/src/i18n.rs
//! Tiny locale-aware string picker.
//!
//! UI/guidance text is localized (currently English + Simplified Chinese);
//! **error messages deliberately stay English** so technical details remain
//! searchable. Language is detected once from `NPT_LANG` (override) or the system
//! locale (`zh*` → Chinese).

use std::sync::OnceLock;

#[derive(Clone, Copy, PartialEq, Eq)]
pub enum Lang {
    En,
    Zh,
}

static LANG: OnceLock<Lang> = OnceLock::new();

pub fn lang() -> Lang {
    *LANG.get_or_init(detect)
}

pub fn is_zh() -> bool {
    lang() == Lang::Zh
}

fn detect() -> Lang {
    if let Ok(v) = std::env::var("NPT_LANG") {
        let v = v.to_lowercase();
        if v.starts_with("zh") {
            return Lang::Zh;
        }
        if v.starts_with("en") {
            return Lang::En;
        }
    }
    match sys_locale::get_locale() {
        Some(l) if l.to_lowercase().starts_with("zh") => Lang::Zh,
        _ => Lang::En,
    }
}

/// Pick a static string by the active language.
pub fn t(en: &'static str, zh: &'static str) -> &'static str {
    if is_zh() {
        zh
    } else {
        en
    }
}

File: src/i18n.ts
//! Minimal bilingual (EN/ZH) UI strings. Language is detected once.
//!
//! Design: user-facing prompts/guidance are localized; low-level error messages
//! from the registry client stay English.

export type Lang = 'en' | 'zh'

const cache: { lang?: Lang } = {}

export function isZh(): boolean {
  return lang() === 'zh'
}

export function lang(): Lang {
  cache.lang ??= detect()
  return cache.lang
}

/** Pick the EN or ZH string for the current language. */
export function t(en: string, zh: string): string {
  return isZh() ? zh : en
}

function detect(): Lang {
  const override = process.env.NPT_LANG?.toLowerCase()
  if (override) {
    if (override.startsWith('zh')) return 'zh'
    if (override.startsWith('en')) return 'en'
  }
  return systemLocale().toLowerCase().startsWith('zh') ? 'zh' : 'en'
}

function systemLocale(): string {
  const env = process.env.LC_ALL || process.env.LC_MESSAGES || process.env.LANG
  if (env) return env
  try {
    return new Intl.DateTimeFormat().resolvedOptions().locale
  } catch {
    return ''
  }
}
